bahamas/app/gateway.py

```diff
@@ -0,0 +1,80 @@
+from bahamas.app import app
+from flask import request, Response
+
+from bahamas.app import db
+from bahamas.app.models import Client, Invoice
+from bahamas.app.validations import *
+
+NEEDED_ARGS = ['email', 'fiscal_id', 'name']
+
+
+@app.route('/store-bahamas-client/<invoice_id>')
```

Comment:
First of all, thank you for submitting your challenge @djguu !

Can you give some insights on REST principles/best practices and how they relate to HTTP methods and routes?

Reply:
Hello!

Yes I will try 😁

So, a REST is a service that allows someone to easily interact using http methods such as GET, POST, PUT, etc. Some best practices is to give a correct response to a call, such as an error, a successfull insertion on a database, etc. Whenever I see a url request such as "/register-client-from-bahamas/1234?name=Bob&email=bob@bob.com", I thinkbof the request as a GET, but now that I had time to rethink what I did, the register should have been a PUT request

Reply:
@djguu show us how would you fix it.

Reply:
@RicCastelhano In order to change this and fix it I would put in line 11
 ```php
@app.route('/store-bahamas-client/<invoice_id>', methods=['PUT'])
```
And upon further examination I found that I needed to remove the `else` in line 42 and pushed the lines underneath to match the `if invoice_q.sec_client is not None:` statements
```php
invoice_q.rel_sec_client = client
response, code = "The client info as been updated", 200
```
Otherwise if the client was already in the invoice as a secondary client, the response would be `"Generic response, something went wrong", 404`, and not the desired `"The client info as been updated", 200`